packages/machine/src/instruction-executor.ts

```diff
@@ -145,8 +163,19 @@ export class InstructionExecutor implements Observable {
   }
 }
 
+export interface IntermediateResults {
+  outbox?: cf.legacy.node.ClientActionMessage;
+  proposedStateTransition?: StateProposal;
+  operation?: ProtocolOperation;
+  signature?: ethers.utils.Signature;
+  inbox?: cf.legacy.node.ClientActionMessage;
```

Comment:
Why are all of these possibly empty?

Reply:
the necessarily start empty before any instructions or middlewares are executed, for example